File: soft/app/types/software.ts
/** 后端 Software JSON（与 Go model 对齐） */
export interface ApiSoftware {
  id: number
  category_id: number
  name: string
  summary: string
  version: string
  cover: string
  cover_thumb: string
  icon: string
  icon_thumb: string
  screenshots: string[]
  size: string
  platforms: string
  website: string
  download_direct: string[]
  download_pan: string[]
  download_extract: string
  published_at?: string
  updated_at_override?: string
  status: number
  created_at: string
  updated_at: string
}

export interface ApiSoftwareVersion {
  id: number
  software_id: number
  version: string
  release_notes: string
  published_at?: string
  download_direct: string[]
  download_pan: string[]
  download_extract: string
  created_at: string
  updated_at: string
}

export interface ApiSoftwareCategory {
  id: number
  name: string
  slug: string
  sort_order: number
  status: number
  created_at: string
  updated_at: string
}

/** 前端展示用 */
export interface SoftwareVersionRow {
  id: string
  version: string
  releaseNotes?: string
  publishedAt?: string
  directDownloads: string[]
  cloudDownloads: string[]
  cloudPassword?: string
}

export interface Software {
  id: string
  name: string
  /** 分类 slug，用于链接 /category/:slug */
  category: string
  categoryLabel: string
  description: string
  version: string
  size: string
  status: 'active' | 'deprecated' | 'beta'
  platform: string[]
  releaseDate?: string
  updateDate?: string
  website?: string
  icon: string
  cover?: string
  screenshots?: string[]
  directDownloads?: string[]
  cloudDownloads?: string[]
  cloudPassword?: string
  /** 详情页：各版本下载 */
  versionRows?: SoftwareVersionRow[]
}

export const statusLabels: Record<string, { label: string; color: 'success' | 'warning' | 'error' }> = {
  active: { label: '正常', color: 'success' },
  deprecated: { label: '已停用', color: 'error' },
  beta: { label: '测试版', color: 'warning' },
}

export const platformIcons: Record<string, string> = {
  Windows: 'i-simple-icons-windows',
  macOS: 'i-simple-icons-apple',
  Linux: 'i-simple-icons-linux',
  Android: 'i-simple-icons-android',
  iOS: 'i-simple-icons-ios',
  Web: 'i-lucide-globe',
}

/** 无后端 slug 映射时的默认图标 */
const SLUG_ICONS: Record<string, string> = {
  productivity: 'i-lucide-zap',
  development: 'i-lucide-code-2',
  design: 'i-lucide-palette',
  system: 'i-lucide-settings',
  media: 'i-lucide-play-circle',
  network: 'i-lucide-wifi',
  dev: 'i-lucide-code-2',
  tool: 'i-lucide-wrench',
}

export function iconForCategorySlug(slug: string): string {
  return SLUG_ICONS[slug] || 'i-lucide-folder'
}

File: 09wp-master/backend/internal/model/software.go
package model

import "time"

// SoftwareCategory 软件分类
type SoftwareCategory struct {
	ID        uint64    `gorm:"primaryKey" json:"id"`
	Name      string    `gorm:"size:100;not null" json:"name"`
	Slug      string    `gorm:"size:120;uniqueIndex;not null" json:"slug"`
	SortOrder int       `gorm:"default:0;index" json:"sort_order"`
	Status    int8      `gorm:"default:1;index" json:"status"` // 1=启用 0=禁用
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

// Software 软件主体
type Software struct {
	ID                uint64         `gorm:"primaryKey" json:"id"`
	CategoryID        uint64         `gorm:"index;not null" json:"category_id"`
	Name              string         `gorm:"size:200;not null;index" json:"name"`
	Summary           string         `gorm:"type:text" json:"summary"`
	Version           string         `gorm:"size:80;default:''" json:"version"`
	Cover             string         `gorm:"size:500;default:''" json:"cover"`
	CoverThumb        string         `gorm:"size:500;default:''" json:"cover_thumb"`
	Icon              string         `gorm:"size:500;default:''" json:"icon"`        // 软件小图标（原图 URL）
	IconThumb         string         `gorm:"size:500;default:''" json:"icon_thumb"` // 列表用缩略图
	Screenshots       JSONStringList `gorm:"type:text;column:screenshots" json:"screenshots"`
	Size              string         `gorm:"size:50;default:''" json:"size"`
	Platforms         string         `gorm:"size:255;default:''" json:"platforms"` // 逗号分隔
	Website           string         `gorm:"size:500;default:''" json:"website"`
	DownloadDirect    JSONStringList `gorm:"type:text;column:download_direct" json:"download_direct"`
	DownloadPan       JSONStringList `gorm:"type:text;column:download_pan" json:"download_pan"`
	DownloadExtract   string         `gorm:"size:100;default:''" json:"download_extract"`
	PublishedAt       *time.Time     `json:"published_at,omitempty"`
	UpdatedAtOverride *time.Time     `json:"updated_at_override,omitempty"` // 业务更新时间
	Status            int8           `gorm:"default:1;index" json:"status"` // 1=上架 0=下架
	CreatedAt         time.Time      `json:"created_at"`
	UpdatedAt         time.Time      `json:"updated_at"`
}

// SoftwareVersion 软件版本
type SoftwareVersion struct {
	ID              uint64         `gorm:"primaryKey" json:"id"`
	SoftwareID      uint64         `gorm:"index;not null" json:"software_id"`
	Version         string         `gorm:"size:80;not null;index" json:"version"`
	ReleaseNotes    string         `gorm:"type:text" json:"release_notes"`
	PublishedAt     *time.Time     `json:"published_at,omitempty"`
	DownloadDirect  JSONStringList `gorm:"type:text;column:download_direct" json:"download_direct"`
	DownloadPan     JSONStringList `gorm:"type:text;column:download_pan" json:"download_pan"`
	DownloadExtract string         `gorm:"size:100;default:''" json:"download_extract"`
	CreatedAt       time.Time      `json:"created_at"`
	UpdatedAt       time.Time      `json:"updated_at"`
}
